torch/_dynamo/variables/dicts.py

```diff
@@ -65,6 +66,8 @@ def is_hashable(x):
         return x.as_proxy().node.meta.get("example_value") is not None
     elif isinstance(x, variables.TupleVariable):
         return all(is_hashable(e) for e in x.items)
+    elif isinstance(x, variables.UserDefinedObjectVariable):
+        return isinstance(x.value, t_Hashable)
```

Comment:
Probably need to check that x.value didn't override `__instancecheck__` otherwise we may execute user code